File: Go/Arrays & Hashing/Valid-Anagram.go
package main

func main() {

}

func isAnagram(s, t string) bool {
	if len(s) != len(t) {
		return false
	}

	frequency := [26]int{}

	for i, sChar := range s {
		tChar := t[i]
		frequency[sChar-'a']++
		frequency[tChar-'a']--
	}

	for _, val := range frequency {
		if val != 0 {
			return false
		}
	}

	return true
}

File: Python/Arrays & Hashing/Valid-Anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        frequency = [0] * 26

        for (s_char, t_char) in zip(s, t):
            frequency[ord(s_char) - ord("a")] += 1
            frequency[ord(t_char) - ord("a")] -= 1

        return len(set(frequency)) == 1
